### ISSUE TITLE
[ISSUE] Issue with creating `databricks_grants` resources

### ISSUE THREAD

**Issue by jl178** [databricks/terraform-provider-databricks#2665]
_Note: I took a look around at similar issues & I couldn't find one that matched this case._

### Configuration
```hcl
terraform {
  required_version = ">= 0.13"
  required_providers {
    databricks = {
      source  = "databricks/databricks"
      version = ">= 1.22.0"
    }
  }
}

locals {
  databricks_metastore_id = "my-cool-metastore-id"
  databricks_account_id = "my-very-cool-account-id"
}
provider "databricks" {
  alias      = "mws"
  host       = "https://accounts.cloud.databricks.com/"
  account_id = local.databricks_account_id
}
provider "databricks" {
  host       = "https://my-cool-workspace.cloud.databricks.com/"
  account_id = local.databricks_account_id
}

resource "databricks_grants" "share_grants" {
  metastore = local.databricks_metastore_id
  grant {
    principal  = "bar"
    privileges = ["USE_RECIPIENT", "USE_SHARE"]
  }
  lifecycle {
    ignore_changes = [
      grant,
    ]
  }
}
```

### Expected Behavior
When **creating** this Terraform resource, I expect it to merge the permissions with the existing ones set on the Databricks platform. For example, if there are already permissions for `foo` to `USE RECIPIENT` and `USE_SHARE`, those should remain intact, and the new permissions for `bar` should be added. Subsequent plans should show the diff between Terraform & Databricks so developers can easily identify what permissions will be removed.

### Actual Behavior
The existing permissions are silently replaced, removing any permissions not defined in the Terraform configuration. This happens due to the `Create` resource using `replacePermissions` (the same as `Update`). [See here.](https://github.com/databricks/terraform-provider-databricks/blob/64f0e73d5c461fd684658a50aea9695d34f3e8ec/catalog/resource_grants.go#L332)

### Steps to Reproduce
1. Define permissions for an object directly in the Databricks platform.
![image](https://github.com/databricks/terraform-provider-databricks/assets/72664723/4558bf6e-6de0-4526-a47a-a1c3c2ecf81c)

2. Define a `databricks_grants` resource in Terraform that sets permissions for the same object but for a different user/group (see above config block).

3. Run `terraform apply`.
```log
Terraform used the selected providers to generate the following execution plan. Resource
actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # databricks_grants.share_grants will be created
  + resource "databricks_grants" "share_grants" {
      + id        = (known after apply)
      + metastore = "my-cool-metastore-id"

      + grant {
          + principal  = "bar"
          + privileges = [
              + "USE_RECIPIENT",
              + "USE_SHARE",
            ]
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

databricks_grants.share_grants: Creating...
databricks_grants.share_grants: Creation complete after 3s [id=metastore/my-cool-metastore-id]
```

4. View the resources for `foo` have been silently replaced in the Databricks UI
![image](https://github.com/databricks/terraform-provider-databricks/assets/72664723/18bdc2d6-3aa4-4dd1-9ab0-ea500945d0d9)

### Terraform and provider versions
```
Terraform v1.3.7
on linux_amd64
+ provider registry.terraform.io/databricks/databricks v1.24.1
```

### Debug Output
https://gist.github.com/jl178/a9ded9b421c0c6f7cce832f6df4b8bf2

### Important Factoids
Our Databricks account is set up into three tiers:
1. Dev
2. Stg
3. Prd

With Terraform, we have one Terraform Cloud workspace **per** SDLC tier. Because of this & combined with the fact that Databricks uses a single Metastore per account, this will cause permissions to break as they are applied across dev -> stg -> prd.

EX:
I give the `TEAM_A-dev` group permissions in the `dev` TF workspace. When I do the same for the next tier (`stg`) `TEAM_A-dev` loses permissions & `TEAM_A-stg` gains permissions.
(Apologies if the above example is confusing)

**_Note: I am willing to attempt a PR to fix this issue myself, if the team is in agreement this behavior is unintended._**

**Comment by jl178** [comment]
This issue also might be impacting the `databricks_permissions` resource as well. I noticed it performs a similar `Update` action. [See here.](https://github.com/databricks/terraform-provider-databricks/blob/64f0e73d5c461fd684658a50aea9695d34f3e8ec/permissions/resource_permissions.go#L472)

This one would be more concerning, since it controls `token` permissions. If token permissions are silently removed, **it will wipe out any tokens created by impacted users/groups.** This could have a large impact if there are many services integrated using tokens.

I am happy to open a separate issue for that with a full test case/debug log/etc, if required.

**Comment by jl178** [comment]
Similar to: https://github.com/databricks/terraform-provider-databricks/issues/2431

**Comment by mgyucht** [comment]
https://github.com/databricks/terraform-provider-databricks/pull/3024 should help with this problem. You'll be able to assign permissions to each group with a different `databricks_grant` block.

**Comment by nkvuong** [comment]
Closed by #3024
